src/streaming/metrics/controllers/MetricsCollectionController.js

```diff
@@ -91,16 +91,21 @@ function MetricsCollectionController(config) {
     }
 
     function setup() {
-
-
         eventBus.on(Events.MANIFEST_UPDATED, update);
-        eventBus.on(Events.STREAM_TEARDOWN_COMPLETE, reset);
+        eventBus.on(Events.STREAM_TEARDOWN_COMPLETE, resetMetricsControllers);
+    }
+
+    function reset() {
+        eventBus.off(Events.MANIFEST_UPDATED, update);
+        eventBus.off(Events.STREAM_TEARDOWN_COMPLETE, resetMetricsControllers);
```

Comment:
Is `resetMetricsControllers` definitely called in all circumstances before `off`? If not, it may be worth calling here?